build/ci/github_actions/operator/wait_for_driver_ready.sh

```diff
@@ -0,0 +1,90 @@
+#!/bin/bash -xe
+set +o pipefail
+
+driver_is_ready=false
+actual_driver_running_time_in_seconds=0
+minimum_driver_running_time_in_seconds=10
+containers_suffix=ibm-block-csi
+declare -a driver_pods_types=(
+  "controller"
+  "node"
+)
+
+get_csi_pods (){
+  kubectl get pod -A -l csi
+}
+
+get_image_pod_by_type (){
+  pod_type=$1
+  container_to_check=$2
+  containers_images=`kubectl get pods $(get_csi_pods | grep $pod_type | awk '{print$2}') -o jsonpath='{range .spec.containers[*]}{.name},{.image} {end}'`
+  for containers_image in $containers_images
+  do
+    if [[  "$containers_image" =~ "$container_to_check," ]]; then
+      echo $containers_image | awk -F , '{print$2}'
+      break
+    fi
+  done
+}
+
+wait_for_driver_pod_to_start (){
+  driver_pod_type=$1
+  while [ "$(get_csi_pods | grep $driver_pod_type | wc -l)" -eq 0 ]; do
+    echo "The $driver_pod_type is not deployed"
+    sleep 1
+  done
+}
+
+wait_for_driver_deployment_to_start (){
+  for driver_pods_type in "${driver_pods_types[@]}"
+  do
+    wait_for_driver_pod_to_start $driver_pods_type
+  done
+}
+
+wait_for_driver_deployment_to_finish (){
+  while [ $driver_is_ready == "false" ]; do
+    if [ "$(get_csi_pods | grep -iv running | grep -iv name | wc -l)" -eq 0 ]; then
+      ((++actual_driver_running_time_in_seconds))
+      if [ $actual_driver_running_time_in_seconds -eq $minimum_driver_running_time_in_seconds ]; then
+        driver_is_ready=true
+      fi
+    else
+      actual_driver_running_time_in_seconds=0
+    fi
+    get_csi_pods
+    sleep 1
+  done
+}
+
+assert_expected_image_in_pod (){
+  pod_type=$1
+  expected_pod_image=$2
+  container_to_check=$containers_suffix-$pod_type
```

Comment:
suffix, or prefix?
what about the sidecars?

Reply:
> > what about the sidecars? should it be in a separate ticket?
> 
> I thought we agreed that we will check only the "main" containers images

do we have a ticket to check the sidecar images?

Reply:
I don't think so, do you want me to create one?

Reply:
yes, thanks

Reply:
CSI-3249

Reply:
> > suffix, or prefix?
> 
> prefix
> done

container, or pod?